### PR TITLE
Split binary functions from math.cc

### PR DESCRIPTION
**Pull request by asi1024**
Part of #7143.

### CODE REVIEW

**Comment by hvy** [comment]
Jenkins, test this please.

File: chainerx_cc/chainerx/routines/binary.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+void CheckBitwiseDtypes(const Array& x1, const Scalar& x2) { CheckBitwiseDtypes(GetKind(x1.dtype()), x2.kind()); }
+
+void CheckInplaceBitwiseDtypes(const Array& x1, const Array& x2) { CheckInplaceBitwiseDtypes(GetKind(x1.dtype()), GetKind(x2.dtype())); }
+
+void CheckInplaceBitwiseDtypes(const Array& x1, const Scalar& x2) { CheckInplaceBitwiseDtypes(GetKind(x1.dtype()), x2.kind()); }
+
+template <typename Impl>
+void BitwiseImpl(const Array& x1, const Array& x2, const Array& out) {
+    CheckEqual(x1.shape(), x2.shape());
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/binary.cc]: This should be an assertion?

**Comment by hvy** [comment]
Basically LGTM. I wrote a small comment, could you take a look?

File: chainerx_cc/chainerx/routines/binary.cc (hunk 1 above)
Comment (asi1024) [reply chainerx_cc/chainerx/routines/binary.cc]: Yes. `internal::BroadcastBinary` broadcasts the inputs beforehand.

**Comment by hvy** [comment]
Moved this ticket back to "Needs Changes" since https://github.com/chainer/chainer/pull/7128/files#r284075612 is unaddressed.

**Comment by hvy** [comment]
Thanks, LGTM. Let me merge this PR once merges are allowed.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Failures seem unrelated, LGTM!

**Pull request merged**